src/features/header/reducers.js

```diff
@@ -0,0 +1,28 @@
+import HEADER_TYPES from "./types";
+
+const initialState = {
+  headerLogoLinks: {}
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+
+    case HEADER_TYPES.LOGO_LINKS.SUCCEEDED:
+      return Object.assign({}, state, {
+        headerLogoLinks: { 
+          logoLinkUrl: action.links[0].extensionParams['logo-link-url'],
+          logoIconUrl: action.links[0].extensionParams['logo-icon-url']
```

Comment:
@jomadoye I don't think the extensionParams is nested under links?  Isn't it just:

action.extensionParams['logo-icon-url']

(For what it's worth, generally the "links" section of our REST responses are links that are REST calls for further details about one or more objects returned in the REST call)

Reply:
Ah my mistake @jomadoye , I see it now... I was just getting confused because there is a "links" object.  It is probably clearer to call it "extensions" since that is what the REST call is returning, a list of extensions.  I'm also fine with starting the "payload" pattern you mentioned.

So I would just change the variable to extensions... but one last thing, I think we need to make sure it checks to see if the array is empty... I think in the default case, actually, where an implementation isn't defining it's own headers, the REST response will just return results: [ ]

Reply:
@mogoodrich If we use the `payload` pattern then there would be no need to call it `extentions` we would end up with `action.payload.extensionParams['logo-link-url']`. so anywhere we see `payload` we immediately know it a data that is coming from the action.

Also regarding the empty array, it was already being handled here(https://github.com/openmrs/openmrs-react-components/pull/6/files#diff-10d86a3b99766f3c48b7bb2d3e67b57fR10). if it returns an empty array, nothing get's saved to the redux store and we end up using the default props value.